### ISSUE TITLE
Remove-Item regression from 7.1.3 - fails to recursively delete dir when path ends with a slash

### ISSUE THREAD

**Issue by rkeithhill** [PowerShell/PowerShell#15492]
## Steps to reproduce

```powershell
(mkdir foo && new-item foo\bar.txt) > $null
Remove-Item -Recurse .\foo\
```

## Expected behavior

I expect this to work as it does wok in v7.1.3

```none
C:\Users\Keith\Code\dotnet
05-29 14:48:34 7> Remove-Item -Recurse .\foo\
C:\Users\Keith\Code\dotnet
```

## Actual behavior

It fails on v7.2.0-preview.5 but only if the path ends with a slash.  The bummer is that PowerShell's tab-completion on the path is what adds the trailing slash.

```none
C:\Users\Keith\Code\dotnet
05-29 14:50:45 94> Remove-Item -Recurse .\foo\
Remove-Item: Cannot remove item C:\Users\Keith\Code\dotnet\foo\: The directory is not empty. : 'C:\Users\Keith\Code\dotnet\foo\'
C:\Users\Keith\Code\dotnet
```

If you remove the trailing slash, v7.2.0-preview.5 works:

```none
C:\Users\Keith\Code\dotnet
05-29 14:53:25 97> Remove-Item -Recurse .\foo
C:\Users\Keith\Code\dotnet
```

## Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.2.0-preview.5
PSEdition                      Core
GitCommitId                    7.2.0-preview.5
OS                             Microsoft Windows 10.0.19043
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by iSazonov** [comment]
I can not reproduce with build from master branch.

**Comment by rkeithhill** [comment]
That's good news.  Does it repro for you in preview.6?  I'm waiting for the PSRL update before I upgrade to preview.6.

**Comment by iSazonov** [comment]
preview.5 and preview.6 from my local builds work well.

**Comment by jborean93** [comment]
Anything special about the files or directories in that path? Could you have a special reparse point that needs special handling?

**Comment by rkeithhill** [comment]
Other than the path is under my home dir, no.  I just tried doing this in my `C:\Temp` dir (no reparse points) and get the same results.

![image](https://user-images.githubusercontent.com/5177512/120225140-ece33080-c201-11eb-8c50-f3e2b88d368f.png)

**Comment by iSazonov** [comment]
Could you please try Preview.6 from zip distributive?

**Comment by rkeithhill** [comment]
OK, tried on another laptop.  preview.5 exhibits the problem there but preview.6 does not.  I think we can close this as fixed.  I wonder if a regression test should be added for this scenario?

**Comment by iSazonov** [comment]
Fixed in #15260 (`string fullPath = Path.TrimEndingDirectorySeparator(fileInfo.FullName);` was added). Dup #15248.
